lib/sales_analyst.rb

```diff
@@ -194,17 +191,40 @@ def unpaid_invoices
     @se.invoices.all.select { |i| !i.is_paid_in_full? }
   end
 
-  # def merchants_ranked_by_revenue
-  #   paid_by_merch = paid_invoices.group_by { |merch| merch.merchant_id }
-  #   paid_by_merch.each_value do |i|
-  #     i.map! { |inv| inv.total }
-  #   end
-  #   totals = paid_by_merch.keys.sort_by do |merch_id|
-  #     paid_by_merch[merch_id].reduce(:+)
-  #   end
-  #   totals.map! do |merch_id|
-  #     @se.merchants.find_by_id(merch_id)
-  #   end.reverse
-  # end
+  def merchants_ranked_by_revenue
+    @se.merchants_by_total_revenue
+  end
+
+  def revenue_by_merchant(merchant_id)
+    @se.revenue_by_merchant(merchant_id)
+  end
+
+  def merchants_with_pending_invoices
+    @se.merchants.all.select do |merchant|
+      merchant.invoices.any? do |invoice|
+        invoice.transactions.all? do |transaction|
+          transaction.result == "failed"
+        end
+      end
+    end
+  end
```

Comment:
@erikaannesmith @Tyjoo26 
To make it a little more readable, I'd pull each of these enumerable blocks into their own methods. So, as the select iterates it calls to a method that then performs the any which then calls a method that preforms the all.